src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -740,6 +740,12 @@ private void AddSegment(int sizeHint = 0)
             _currentSegment = owner.Memory;
             _currentSegmentOwner = owner;
         }
+        else if (sizeHint <= MemoryPool<byte>.Shared.MaxBufferSize)
```

Comment:
What are the values of `_memoryPool.MaxBufferSize` and `MemoryPool<byte>.Shared.MaxBufferSize`? How much of a gap is there between these values?

Reply:
4k vs 2GB